src/separateLater/Sentinel1/Metadata/Sentinel1Metadata.jl

```diff
@@ -0,0 +1,558 @@
+    slantRangeTime = [parse(Float64, elem["slantRangeTime"]) for elem in geolocation]
+    elevationAngle = [parse(Float64, elem["elevationAngle"]) for elem in geolocation]
+    incidenceAngle = [parse(Float64, elem["incidenceAngle"]) for elem in geolocation]
+    height = [parse(Float64, elem["height"]) for elem in geolocation]
+
+    geolocationgrid = GeolocationGrid(lines,
+        samples,
+        latitude,
+        longitude,
+        azimuthTime,
+        slantRangeTime,
+        elevationAngle,
+        incidenceAngle,
+        height)
+    return geolocationgrid
+end
+
+""""
+Burst:
+    For Sentinel-1
+    Getting Burst data for a swath/polarization.
+    
+Inputs:
+    - metaDict, linesPerBurst, azimuthFrequency, startSensingTime
+
+
+    Note to self:
+        This should be cleaned up greatly.
+"""
+
+function Burst(metaDict, linesPerBurst::Int64, azimuthFrequency::Float64, startSensingTime::Dates.DateTime)::Burst
+    #dict with burst info
+    burst = metaDict["product"]["swathTiming"]["burstList"]["burst"]
+    #dict with dopplercentroid estiamted
+    dopplerCentroid = metaDict["product"]["dopplerCentroid"]["dcEstimateList"]["dcEstimate"]
+    # number of burst.
+    burstCount = parse(Int, metaDict["product"]["swathTiming"]["burstList"][:count])
+
+
+    #data from the dict
+    azimuthTime = [Dates.DateTime(elem["azimuthTime"][1:23]) for elem in burst]
+    azimuthAnxTime = [parse(Float64, elem["azimuthAnxTime"]) for elem in burst]
+    sensingTime = [Dates.DateTime(elem["sensingTime"][1:23]) for elem in burst]
+    byteOffset = [parse(Int64, elem["byteOffset"]) for elem in burst]
+    firstValidSample = [parse.(Int, split(elem["firstValidSample"][""])) for elem in burst]
+    lastValidSample = [parse.(Int, split(elem["lastValidSample"][""])) for elem in burst]
+    burstId = [parse.(Int64, split(elem["burstId"][""])[1]) for elem in burst]
+
+    absoluteBurstId = [parse.(Int64, split(elem["burstId"][:absolute])[1]) for elem in burst]
+    #   burst times are calculte as time differece between sensing start
+
+    burstTimes = getTimeDifference.(startSensingTime, azimuthTime)
+    burstMidTimes = burstTimes .+ linesPerBurst / (2 * azimuthFrequency)
+
+
+
+    #select the polynomials and t0's closest to mid burst time
+    data_dc_polynomial = Array{Float64,2}(undef, burstCount, 3)
+    data_dc_t0 = Array{Float64,1}(undef, burstCount)
+
+    dcTimeDifference = Array{Float64,1}(undef, length(dopplerCentroid))
+
+
+    for i in 1:burstCount
+        for j in 1:length(dopplerCentroid)
+            dcTime = getTimeDifference(startSensingTime, Dates.DateTime(dopplerCentroid[j]["azimuthTime"][1:23]))
+            dcTimeDifference[j] = abs.(dcTime - burstMidTimes[i])
+        end
+        best_dc_index = argmin(dcTimeDifference)
+        data_dc_polynomial[i, :] = [parse(Float64, param) for param in split(dopplerCentroid[best_dc_index]["dataDcPolynomial"][""])]
```

Comment:
Split this up in 2 lines
`best_polynomial = split(dopplerCentroid[best_dc_index]["dataDcPolynomial"][""])`
`data_dc_polynomial[i, :]= parse.(Float64, best_polynomial) `

Maybe look up the `.` notation in Julia. It is very useful when you just want to apply a simple function on all elements in an array